libraries/AP_PiccoloCAN/piccolo_protocol/scaledencode.c

```diff
@@ -377,15 +377,15 @@ void float32ScaledTo2SignedBeBytes(float value, uint8_t* bytes, int* index, floa
     // Make sure number fits in the range
     if(scaledvalue >= 0)
     {
-        if(scaledvalue >= 32767)
-            number = 32767;
+        if(scaledvalue >= INT16_MAX)
+            number = INT16_MAX;
         else
             number = (int16_t)(scaledvalue + 0.5f); // account for fractional truncation
     }
     else
     {
-        if(scaledvalue <= (-32767 - 1))
-            number = (-32767 - 1);
+        if(scaledvalue <= (INT16_MIN))
+            number = (INT16_MIN);
```

Comment:
This doesn't look right - and we don't want to modify this code as it's not from our tree anyway

Reply:
I agree with you.